client/state/selectors/is-coming-soon-site.js

```diff
@@ -0,0 +1,35 @@
+/**
+ * Internal dependencies
+ */
+
+import getRawSite from 'state/selectors/get-raw-site';
+import { getSiteSettings } from 'state/site-settings/selectors';
+import isPrivateSite from 'state/selectors/is-private-site';
+
+/**
+ * Returns true if the site is coming_soon
+ *
+ * @param {object} state Global state tree
+ * @param {object} siteId Site ID
+ * @returns {boolean} True if site is coming_soon
+ */
+export default function isComingSoonSite( state, siteId ) {
```

Comment:
Minor: I'd change these names around in the function & file names (Is Site Coming Soon vs. Is Coming Soon Site).  It just reads better to me, personally. 

```suggestion
export default function isSiteComingSoon( state, siteId ) {
```

Reply:
I agree, I only went for `isComingSoonSite` to be consistent with `isPrivateSite`. I'd like both better is `isSiteCominigSoon` and `isSitePrivate` though!

Reply:
I started to make this change (and the `isSitePrivate` one), but it's a slippery slope. We follow both conventions in our code 🙄(i.e. `isJetpackSite` or `isSiteStore`). I do think that `is-site-` makes for a cleaner file structure.

I'll push the changes, but I don't mind if we just drop the commits if we decide against that approach.

Reply:
I opted not to finish the change for `isPrivateSite` to `isSitePrivate`. It's a pretty large undertaking and will junk up this PR. If we decide we want to pursue the change, we can do it in a follow-up.

Reply:
👍  Thank you!